cpp/src/utilities/device_operators.cuh

```diff
@@ -76,15 +76,35 @@ struct DeviceCount {
 
 /* @brief binary `min` operator */
 struct DeviceMin {
-  template <typename T>
+  template <typename T,
+            typename std::enable_if_t<!std::is_same<T, cudf::string_view>::value>* = nullptr>
   CUDA_HOST_DEVICE_CALLABLE T operator()(const T& lhs, const T& rhs) {
     return std::min(lhs, rhs);
   }
 
-  template <typename T>
+  template <typename T,
+            typename std::enable_if_t<!std::is_same<T, cudf::string_view>::value>* = nullptr>
   static constexpr T identity() {
     return std::numeric_limits<T>::max();
   }
+
+  template <typename T,
+            typename std::enable_if_t<std::is_same<T, cudf::string_view>::value>* = nullptr>
+  CUDA_HOST_DEVICE_CALLABLE T operator()(const T& lhs, const T& rhs) {
+    if (lhs.empty())
```

Comment:
Why does this require special logic? `string_view::operator<` is already defined, so I wouldn't think it should require any special handling. It looks like you want empty strings ordered after all other strings? I would think empty strings come _before_ all other strings. Either way, this logic should go into `string_view::operator<` and _not_ here. CC @davidwendt

Reply:
For string column with nulls along with null replacement iterator, we need to provide identity for `min` operator. Largest string_view is not possible, so i added special handling of null string_view. 
`empty()` checking seems wrong. I will change it to `is_null()`.

Reply:
Okay, yes, definitely use `is_null()` instead and make sure to clearly document why a null `string_view` is the identity.

Reply:
@jrhemstad @davidwendt I hear you discussed using invalid UTF8 characters as a sentinel instead. Can you elaborate?

Reply:
We either need to find a sentinel `string_view` that is larger than all other strings, or refactor the algorithms to avoid the need for an identity. Either way, it should not rely on a null string_view. The concept of a "null" `string_view` is problematic for the reasons I outlined here: https://github.com/rapidsai/cudf/issues/3346

Reply:
This was resolved, and yet a null `string_view` is still being used for the identity of _both_ `min/max`. This is problematic.

Reply:
I resolved this because I assumed, once we have a solution for null string_view, it will be fixed as part of #3346.

Reply:
Alternate idea is to write template specialization for string_view. use `make_pair_iterator` instead of `make_null_replacement_iterator`.  Write specialization of `deviceMin`, `deviceMax` for `thrust::pair<string_view, bool>`. 
and finally convert `thrust::pair<string_view, bool>` to `string_scalar`. But it should be another PR after pair iterator PR https://github.com/rapidsai/cudf/pull/3490 is merged.

Reply:
I think #3346 is a separate issue. What we need to move this PR forward is a sentinel `string_view` value other than a NULL one to use for an identity for `min`. @davidwendt

Reply:
I'm inclined to agree with pair-iterator-specialization approach.
Alternatively, we can create an artificial sentinel for min perhaps and I'm hoping an empty string can be used as sentinel/identity for max. 
A string with the following bytes `"\xF7\xBF\xBF\xBF"` in device memory is the highest possible valid UTF-8 encoded character. This could always be subverted by making a longer version of this string but the appearance of this character is unlikely to happen.
```
  string_scalar max_scalar("\xF7\xBF\xBF\xBF");
  string_view max_str(max_scalar.data(), max_scalar.size());
```
If this proves insufficient, we could rework this in a later PR.

Reply:
The pair iterator approach won't work because there'd be no way to distinguish a real null string element from the sentinel value.

Reply:
> The pair iterator approach won't work because there'd be no way to distinguish a real null string element from the sentinel value.

@jrhemstad  by 'real null string', you mean empty string?

Reply:
For pair iterator approach, there is no sentinel value.

Reply:
Reductions are not the only feature that make use of the `DeviceMin/DeviceMax` operators, and as such, they cannot be forced to use the pair iterator approach. 

> @jrhemstad by 'real null string', you mean empty string?

No, I mean a string element at index `i` where `is_null(i) == true`.

> we can create an artificial sentinel for min

@karthikeyann please work with @davidwendt to define a `string_view` sentinel for `min`.

It should look something like:
```c++
constexpr string_view("...",n);
```

Reply:
sentinel value for deviceMin added. It works now. so resolving this comment.